src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractSyntaxFactsService.cs

```diff
@@ -542,11 +542,22 @@ public bool IsOnHeader<THoleSyntax>(int position, SyntaxNode ownerOfHeader, Synt
 
         protected int GetStartOfNodeExcludingAttributes(SyntaxNode node)
         {
-            var attributeLists = GetAttributeLists(node);
-            var start = attributeLists.LastOrDefault()?.GetLastToken().GetNextToken().SpanStart ??
-                        node.SpanStart;
+            var attributeList = GetAttributeLists(node);
+            if (attributeList.Any())
+            {
+                var endOfAttributeLists = attributeList.Last().Span.End;
+                var afterAttributesToken = node.FindTokenOnRightOfPosition(endOfAttributeLists);
+
+                var endOfNode = node.Span.End;
+                var startOfTokenAfterAttributes = afterAttributesToken.Span.Start;
+                var startOfNodeWithoutAttributes = endOfNode >= startOfTokenAfterAttributes
+                    ? afterAttributesToken.Span.Start
+                    : endOfNode;
+
+                return startOfNodeWithoutAttributes;
```

Comment:
startOfNodeWithoutAttributes == afterAttributeToken.Span.Start why not use it all places?
also, startOfNodeWithoutAttrubuites == Math.Min(startofTokenAfterAttributes, endOfNode)?

Reply:
whole code seems could be simplified to
return math.min(afterAttributesToken.Span.Start, node.Span.End)

Reply:
ah. it is actually reserved. sorry.

Reply:
Copying and refactoring lead to this, will simplify.